src/esp/gfx/RenderCamera.cpp

```diff
@@ -15,6 +15,41 @@ namespace Mn = Magnum;
 namespace esp {
 namespace gfx {
 
+struct CullingResult {
+  // the culling result
+  bool intersected = true;
+  // the frustum plane that culled the object
+  int frustumPlaneIndex = 0;
+};
+
+/**
+ * @brief do frustum culling with temporal coherence
+ * @param range, the axis-aligned bounding box
+ * @param frustum, the frustum
+ * @param frustumPlaneIndex, the frustum plane in last frame that culled the
+ * aabb (default: 0)
+ */
+CullingResult rangeFrustum(const Mn::Range3D& range,
```

Comment:
I'd prefer using an https://doc.magnum.graphics/corrade/classCorrade_1_1Containers_1_1Optional.html here.

Reply:
Can I ask why?

Reply:
Doesn't requiring introducing an additional type.

Reply:
I see. You were actually talking the return type. I thought you were referring the arguments.

That is a good idea.

Reply:
Ah, apologies for lack of clarity.

Reply:
Not a problem at all. Good advice, makes the code simpler. Will change.